src/Compilers/CSharp/Portable/Lowering/LocalRewriter_NullChecking.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        internal BoundStatement RewriteNullChecking(ImmutableArray<BoundStatement> statements, ImmutableArray<LocalSymbol> locals)
+        {
+            if (_factory.CurrentFunction.Parameters.Any(x => x is SourceParameterSymbolBase param
+                                                             && param.IsNullChecked))
```

Comment:
I probably suggested adding this check in an earlier iteration, but with the current approach it seems simpler to merge this with the `AddNullChecksToBody` below so there's only check.

In short, we could combine the two methods with something like:
```C#
ArrayBuilder<BoundStatement> statementList = null;
foreach (var parameter in _factory.CurrentFunction.Parameters)
{
    if (param.IsNullChecked && ...)
    {
        statementList ??= ArrayBuilder<BoundStatement>.GetInstance();
        statementList.Add(...);
    }
}
if (statementList is null) return body;
...
```